### PR TITLE
Add support to create an OIDC app without the callback URLs in the manual setup

### PR DESCRIPTION
**Pull request by NipuniBhagya**
### Purpose

- This PR introduces the capability to add OIDC protocol to an application without specifying the callback URL if a grant type that doesn't require a callback URL is configured.
- This also improves the application edit section to update the application protocol without specifying the callback URL if the configured grant type doesn't use a callback URL.
- Resolves https://github.com/wso2/product-is/issues/10221

### CODE REVIEW

File: apps/console/src/features/applications/components/wizard/application-create-wizard.tsx [hunk 1]
```diff
@@ -270,7 +270,8 @@ export const ApplicationCreateWizard: FunctionComponent<ApplicationCreateWizardP
      *
      * @param values - Form values.
      */
-    const HandleApplicationProtocolsUpdate = (values: any): void => {
+    const handleApplicationProtocolsUpdate = (values: any): void => {
+        console.log("Values", values);
```
Comment (jeradrutnam) [inline apps/console/src/features/applications/components/wizard/application-create-wizard.tsx:274]:
```suggestion
```

Remove the console.log

File: apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx [hunk 2]
```diff
@@ -1271,22 +1329,26 @@ export const InboundOIDCForm: FunctionComponent<InboundOIDCFormPropsInterface> =
...
-                            }
-                        });
+                        if (showCallbackURLField) {
+                            submitUrl((url: string) => {
+                                if (isEmpty(callBackUrls) && isEmpty(url)) {
+                                    setShowURLError(true);
+                                    scrollToInValidField("url");
+                                } else {
+                                    submitOrigin((origin) => {
+                                        // TODO: Remove the empty check when the backend is fixed.
```
Comment (jeradrutnam) [inline apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:1339]: Shall we create an issue and refer it here?

Comment (NipuniBhagya) [reply apps/console/src/features/applications/components/forms/inbound-oidc-form.tsx:1339]: Added the reference to the git issue with bdfcb09

File: apps/console/src/features/applications/components/wizard/oauth-protocol-settings-wizard-form.tsx [hunk 3]
```diff
@@ -102,23 +119,76 @@ export const OauthProtocolSettingsWizardForm: FunctionComponent<OAuthProtocolSet
     const [ showRefreshToken, setShowRefreshToken ] = useState(false);
     const [ showURLError, setShowURLError ] = useState(false);
     const [ callbackURLsErrorLabel, setCallbackURLsErrorLabel ] = useState<ReactElement>(null);
-    // TODO enable after fixing callbackURL.
-    // const [showCallbackUrl, setShowCallbackUrl] = useState(false);
+    const [ showCallbackURLField, setShowCallbackURLField ] = useState(true);
```
Comment (brionmario) [inline apps/console/src/features/applications/components/wizard/oauth-protocol-settings-wizard-form.tsx:122]:
```suggestion
    const [ showCallbackURLField, setShowCallbackURLField ] = useState<boolean>(true);
```

File: apps/console/src/features/applications/components/wizard/oauth-protocol-settings-wizard-form.tsx [hunk 4]
```diff
@@ -102,23 +119,76 @@ export const OauthProtocolSettingsWizardForm: FunctionComponent<OAuthProtocolSet
...
 
     // Maintain the state if the user allowed the CORS for the
     // origin of the configured callback URL(s).
     const [ allowCORSUrls, setAllowCORSUrls ] = useState<string[]>([]);
 
+    /**
+     * Show the grant types only for the custom protocol template.
+     */
+    useEffect(() => {
+        if (selectedTemplate?.id === "default-oidc") {
```
Comment (brionmario) [inline apps/console/src/features/applications/components/wizard/oauth-protocol-settings-wizard-form.tsx:136]: Use the enum here.

```suggestion
        if (selectedTemplate?.id === DefaultProtocolTemplate.OIDC) {
```

**Review by brionmario — APPROVED** [review]

**Pull request merged**